homeassistant/components/axis/config_flow.py

```diff
@@ -136,7 +136,7 @@ async def _create_entry(self):
         title = f"{model} - {self.serial}"
         return self.async_create_entry(title=title, data=self.device_config)
 
-    async def async_step_reauth(self, device_config: dict):
+    async def async_step_reauth(self, device_config: dict) -> FlowResult:
```

Comment:
I'm sure that the type hint for device_config could also be improved, but there are so many possibilities for `async_step_***` that I'm not sure it would be useful to implement via the pylint plugin.
I think just enforcing the FlowResult is already an improvement...

Reply:
I went ahead and adjusted the generic `dict` also, at least in this PR...